librz/analysis/fcn.c

```diff
@@ -521,12 +520,29 @@ static inline bool does_arch_destroys_dst(const char *arch) {
 	return arch && (!strncmp(arch, "arm", 3) || !strcmp(arch, "riscv") || !strcmp(arch, "ppc"));
 }
 
-static int fcn_recurse(RzAnalysis *analysis, RzAnalysisFunction *fcn, ut64 addr, ut64 len, int depth) {
+static int analyze_function_locally(RzAnalysis *analysis, RzAnalysisFunction *fcn, ut64 address) {
+	RzVector tasks;
+	rz_vector_init(&tasks, sizeof(RzAnalysisTaskItem), NULL, NULL);
+	RzAnalysisTaskItem item = { fcn, NULL, fcn->stack, address };
+	rz_vector_push(&tasks, &item);
+	int saved_stack = fcn->stack; // TODO: DO NOT use fcn->stack to keep track of stack during analysis
+	int ret = rz_analysis_run_tasks(&tasks);
+	rz_vector_fini(&tasks);
+	fcn->stack = saved_stack;
+	return ret;
+}
+
+static int run_basic_block_analysis(RzAnalysisTaskItem *item, RzVector *tasks) {
+	RzAnalysis *analysis = item->fcn->analysis;
```

Comment:
Please add also `rz_return_val_if_fail(item && tasks)`.
Also, should we exit quickly if `tasks` vector is empty?

Reply:
> Also, should we exit quickly if tasks vector is empty?

@XVilka No, here the work is being done for `item`, the `tasks` is used to add new items to analyze later in case of jump/switch tables